main.rb

```diff
@@ -0,0 +1,108 @@
+# rubocop:disable all
+
+require './app'
+class Main
+    def initialize
+      @app = App.new
+      puts 'Welcome to School Library App!'
+    end
+  
+    def menu
+      puts %(
+  Please choose an option by enterin a number:
+  1 - List all books
+  2 - List all people
+  3 - Create a person
+  4 - Create a book
+  5 - Create a rental
+  6 - List all rentals for a given person id
+  7 - Exit)
+    end
+  
+    def create_person
+      print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+      person_type = gets.chomp
+      unless %w[1 2].include?(person_type)
+        puts 'Invalid option'
+        return
+      end
+      print 'Age: '
+      age = gets.chomp
+      print 'Name: '
+      name = gets.chomp
+      case person_type
+      when '1'
+        print 'Has parent permission? [Y/N]: '
+        @app.add_student(age, name, gets.chomp.downcase == 'y')
+      when '2'
+        print 'Specialization: '
+        @app.add_teacher(age, gets.chomp, name)
+      end
+      puts 'Person created successfully'
+    end
+  
+    def create_book
+      print 'Title: '
+      title = gets.chomp
+  
+      print 'Author: '
+      author = gets.chomp
+  
+      @app.add_book(title, author)
+      puts 'Book created successfully'
+    end
+  
+    def create_rental
+      puts 'Select a book from the following list by number'
+      @app.list_books
+      book_id = gets.chomp.to_i
+      puts 'Select a person from the following list by number (not id)'
+      @app.list_people(with_number: true)
+      people_id = gets.chomp.to_i
+      print 'Date: '
+      date = gets.chomp
+      rental = @app.add_rental(date, book_id, people_id) ? 'Rental created successfully' : 'Invalid book or person ID'
```

Comment:
- [ ] After successfully creating a teacher and book, when I try to create a rental, this error comes up, please fix it

<img width="1134" alt="Screenshot 2023-07-01 at 17 11 20" src="https://github.com/Daniel-Kigozi/school_library/assets/64774936/63bdfc07-9e59-48e1-b055-e1818c8d4990">